engine/storage/volume/src/main/java/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java

```diff
@@ -732,7 +734,15 @@ protected void createVolumeFromBaseImageAsync(VolumeInfo volume, DataObject temp
         caller.setCallback(caller.getTarget().createVolumeFromBaseImageCallBack(null, null));
         caller.setContext(context);
 
-        motionSrv.copyAsync(context.templateOnStore, volumeOnPrimaryStorage, caller);
+        if (!volume.isDeployAsIs()) {
+            motionSrv.copyAsync(context.templateOnStore, volumeOnPrimaryStorage, caller);
+        } else {
+            Answer answer = new Answer(null);
+            CopyCommandResult result = new CopyCommandResult(null, null);
+            result.setSuccess(true);
+            caller.complete(result);
```

Comment:
I must say this is a terrible hack, maybe we should rename the method to reflect that the copy may not even be attempted?